lib/devise/models/database_authenticatable.rb

```diff
@@ -37,6 +37,7 @@ def self.required_fields(klass)
 
       # Generates password encryption based on the given value.
       def password=(new_password)
+        self.encrypted_password_will_change! #trigger validations of the newly set password
```

Comment:
This is outside the @password.present? so that if a blank password is set it still triggers validation, without it here tests fail.